layout/new.go

```diff
@@ -58,6 +63,67 @@ func NewImage(path string, ops ...imgutil.ImageOption) (*Image, error) {
 	}, nil
 }
 
+// NewIndex will return an OCI ImageIndex saved on disk using OCI media Types. It can be modified and saved to a registry
+func NewIndex(repoName, path string, ops ...imgutil.IndexOption) (idx *ImageIndex, err error) {
```

Comment:
The only option that we actually use in this function is `idxOps.BaseImageIndexRepoName`. What if this function just returned a `v1.ImageIndex` (instead of wrapping it in `CNBIndex`)? That way the caller could use this to construct an index to provide to `imgutil.NewCNBIndex`.

Reply:
Makes sense, apparently we don't have any particular behavior for `layout` or `remote` package so wrapping the `CNBIndex` is not adding any value right now.

Reply:
I started to do this in https://github.com/husni-faiz/imgutil/pull/12 but then I saw some value in at least returning a CNB image from the constructor, since we are formatting the options to the extent that we set the base index "instance". Also have it work the same way as the image constructor is convenient